cloud/amazon/ec2_eni.py

```diff
@@ -369,7 +414,7 @@ def main():
 
     if region:
         try:
-            connection = connect_to_aws(boto.ec2, region, **aws_connect_params)
+            connection = connect_to_aws(boto.vpc, region, **aws_connect_params)
```

Comment:
get_all_subnets, used to determine which vpc the subnet is in needs a vpc connection rather than an ec2 connection